### PR TITLE
map state for endpoints

### PR DESCRIPTION
**Pull request by amit-shah-cb**
This PR maps db results to client consuming types as defined in hooks/types.ts

### CODE REVIEW

File: src/app/api/profile/score/route.ts [hunk 1]
```diff
@@ -37,24 +38,52 @@ export async function GET(req: Request) {
...
-                 currentLevel = levels[nextLevelIdx-1];
-             }
-        }else{
-            nextLevel = levels[0];
-        }
+          }); 
+        
+        if (score){
+          const nextLevelIdx = levels.findIndex((level:any)=>level.threshold_points>score.current_score);
+          var nextLevel = null;
```
Comment (siwan-cb) [inline src/app/api/profile/score/route.ts:45]: no vars. use `let`

File: src/app/api/profile/score/route.ts [hunk 2]
```diff
@@ -37,24 +38,52 @@ export async function GET(req: Request) {
...
+          var currentLevel = null;
+          if (nextLevelIdx!=null){
+              nextLevel = levels[nextLevelIdx];
+              if (nextLevelIdx>0){
```
Comment (siwan-cb) [inline src/app/api/profile/score/route.ts:49]: do you have eslint configured for your IDE?

File: src/app/api/profile/score/route.ts [hunk 3]
```diff
@@ -37,24 +38,52 @@ export async function GET(req: Request) {
...
+                  currentLevel = levels[nextLevelIdx-1];
+              }
+          }
+      }else{
+        nextLevel = levels[0];
+      }
+    
+    console.log("Map Score")
```
Comment (siwan-cb) [inline src/app/api/profile/score/route.ts:57]: remove console.log

File: src/app/api/profile/score/route.ts [hunk 4]
```diff
@@ -37,24 +38,52 @@ export async function GET(req: Request) {
...
+    const scoreState = mapToScore(
+      currentLevel,
+      nextLevel,
+      score
+    )
+    return NextResponse.json(scoreState);
+}
 
+function mapToScore(c:any,n:any,s:any):ScoreState{
+  console.log("here");
```
Comment (siwan-cb) [inline src/app/api/profile/score/route.ts:67]: remove console.log

File: src/app/api/profile/score/route.ts [hunk 5]
```diff
@@ -37,24 +38,52 @@ export async function GET(req: Request) {
...
+    console.log("Map Score")
+    const scoreState = mapToScore(
+      currentLevel,
+      nextLevel,
+      score
+    )
+    return NextResponse.json(scoreState);
+}
 
+function mapToScore(c:any,n:any,s:any):ScoreState{
```
Comment (siwan-cb) [inline src/app/api/profile/score/route.ts:66]: no `any`

File: src/app/api/profile/state/route.ts [hunk 6]
```diff
@@ -1,16 +1,45 @@
...
+        if (!addressSet.has(key)) {
+            addressSet.add(key);
+            results.push(mapToBadge(d));
+        }
+    }
+    return NextResponse.json(results);
 }
+
+
+function mapToBadge(b:any):Badge {
```
Comment (siwan-cb) [inline src/app/api/profile/state/route.ts:30]: no `any`

Comment (amit-shah-cb) [reply src/app/api/profile/state/route.ts:30]: The result from the rawQuery is anonymous type

File: src/app/api/profile/score/route.ts (hunk 4 above)
Comment (taycaldwell) [inline src/app/api/profile/score/route.ts]: Remove if debug console.log

**Review by taycaldwell — APPROVED** [review]

**Pull request merged**